### PR TITLE
[HttpClient] Add documentation for Retryable client

### LINKED ISSUE #14252: [HttpClient] Added RetryHttpClient
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#38182
| PR author(s) | @jderusse
| Merged in    | 5.2-dev

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #14252, #14317

### CODE REVIEW

File: http_client.rst [hunk 1]
```diff
@@ -654,6 +655,26 @@ making a request. Use the ``max_redirects`` setting to configure this behavior
         'max_redirects' => 0,
     ]);
 
+Retry Failed Requests
+~~~~~~~~~~~~~~~~~~~~~
+
+Some times, requests failed because of temporary issue in the server or
+because network issue. You can use the
+:class:`Symfony\\Component\\HttpClient\\RetryableHttpClient`
+client to automatically retry the request when it failed.::
```
Comment (OskarStark) [inline http_client.rst:664]: it fails::

File: http_client.rst [hunk 2]
```diff
@@ -654,6 +655,26 @@ making a request. Use the ``max_redirects`` setting to configure this behavior
...
+
+    $client = new RetryableHttpClient(HttpClient::create());
```
Comment (OskarStark) [inline http_client.rst:666]: Could you please add a use statement for the new client? Thanks

File: http_client.rst [hunk 3]
```diff
@@ -654,6 +655,26 @@ making a request. Use the ``max_redirects`` setting to configure this behavior
...
+
+The ``RetryableHttpClient`` uses a
+:class:`Symfony\\Component\\HttpClient\\Retry\\RetryDeciderInterface` to
+decide if the request should be retried, and a
+:class:`Symfony\\Component\\HttpClient\\Retry\\RetryBackOffInterface` to
+define the waiting time between each request.
+
+By default, it retries until 3 attemps, the requests responding with a
+status code in (423, 425, 429, 500, 502, 503, 504, 507 or 510) and wait
```
Comment (OskarStark) [inline http_client.rst:675]: Can we say steht like "... for 4xx and 5xx status codes..." or is it not the case for all 400' and 500' codes?

Comment (jderusse) [reply http_client.rst:675]: it's exactly those http status code (for instance 404 is not retried).
But if it's too verbose, we can reword the sentence `the requests responding with a temporary failure status code`. What do you think?

File: http_client.rst (hunk 2 above)
Comment (jderusse) [reply http_client.rst:666]: done

File: http_client.rst (hunk 3 above)
Comment (javiereguiluz) [reply http_client.rst:677]: @jderusse when using the HttpClient inside a Symfony app, how can I enable this feature but rely on all its default values? Would the following be enough?

```yaml
    # config/packages/framework.yaml
    framework:
        http_client:
            retry_failed: null
```

Comment (jderusse) [reply http_client.rst:677]: @javiereguiluz yes both works. 

This is `canBeEnabled` configuration.
default `disabled`, set `null`, `true`, a `enabled: true` subkey, or anyother subkey to enable it.

File: http_client.rst [hunk 4]
```diff
@@ -654,6 +655,28 @@ making a request. Use the ``max_redirects`` setting to configure this behavior
...
+because network issue. You can use the
+:class:`Symfony\\Component\\HttpClient\\RetryableHttpClient`
+client to automatically retry the request when it fails.::
+
+    use Symfony\Component\HttpClient\RetryableHttpClient;
+
+    $client = new RetryableHttpClient(HttpClient::create());
+
+The ``RetryableHttpClient`` uses a
+:class:`Symfony\\Component\\HttpClient\\Retry\\RetryDeciderInterface` to
```
Comment (jderusse) [inline http_client.rst:671]: Do you think we need a section to explain how user can implement their own Decider and BackOff ?

Comment (javiereguiluz) [reply http_client.rst:671]: It's not necessary because we no longer document components on its own, but when used inside a Symfony app. Thanks!

**Comment by javiereguiluz** [comment]
Jérémy, thanks a lot for implementing this feature and contributing its docs. We've just merged. We did some rewords to explain first the integration with Symfony and we added the `versionadded` directives. We also removed all the *"XXX option can't be set if you also set YYY option"* because that will be displayed by the application in easy to understand error messages, so there's no need to complicate the docs with that. Thanks!

**Pull request merged**